runtime/near-vm-runner/src/imports.rs

```diff
@@ -469,21 +452,28 @@ pub(crate) mod wasmtime {
               $func:ident < [ $( $arg_name:ident : $arg_type:ident ),* ] -> [ $( $returns:ident ),* ] >
             ) => {
                 #[allow(unused_parens)]
-                fn $func( $( $arg_name: rust2wasm!($arg_type) ),* ) -> Result<($( rust2wasm!($returns)),*), wasmtime::Trap> {
+                fn $func(caller: wasmtime::Caller<'_, ()>, $( $arg_name: $arg_type ),* ) -> Result<($( $returns ),*), wasmtime::Trap> {
                     const IS_GAS: bool = str_eq(stringify!($func), "gas");
                     let _span = if IS_GAS {
                         None
                     } else {
                         Some(tracing::trace_span!(target: "host-function", stringify!($func)).entered())
                     };
+                    // the below is bad. don't do this at home. it probably works thanks to the exact way the system is setup.
+                    // Thanksfully, this doesn't run in production, and hopefully should be possible to remove before we even
+                    // consider doing so.
                     let data = CALLER_CONTEXT.with(|caller_context| {
                         unsafe {
                             *caller_context.get()
                         }
                     });
+                    unsafe {
+                        // transmute the lifetime of caller so it's possible to put it in a thread-local
```

Comment:
I presonally have a soft preference for grammar and punctuation in coments: 

```
// Transmute the lifetime of caller so it's possible to put it in a thread-local.
```

Not because it makes the code that much more readable, but because it just encourages you to write better comments (writing a proper paragraph is serious business, while a mere comment is just that).